### ISSUE TITLE
choose color with palette options "lose" focus

### ISSUE THREAD

**Issue by GuillaumeGrandjean** [Jimdo/angular-spectrum-colorpicker#53]
When I chose a color palette with, it works perfectly, the 2nd test, the cursor returns to the beginning of the line, I do not understand at all ... 

Another person had the same problem, but had no response:
http://plnkr.co/edit/xSA66CWfeXQMMzLlJ3Tm?p=preview

Please help me out on this.

**Comment by Xiphe** [comment]
hey thanks for the plunker. i can reproduce the issue. strange thing
not sure if i have time in the next future but i have it on my list

**Comment by Xiphe** [comment]
could you add a reference to "the other person with the same problem"?

**Comment by GuillaumeGrandjean** [comment]
Thanks for your answer.
http://stackoverflow.com/questions/36762189/text-angular-color-picker-spectrum-pallet-issue here you can find the issue of the other person.

I tried to contact him, but no answer :(

**Comment by Xiphe** [comment]
from what I see, the plunker you submitted and that is from the referred stackoverflow issue contains two colorpickers. the first one is from this lib and works fine, the second one is from [ruhley/angular-color-picker](https://github.com/ruhley/angular-color-picker) and resets the input.

closing here since i assume this is not caused by our library

**Comment by GuillaumeGrandjean** [comment]
Sorry but don't understand, you said 

`taRegisterTool('fontColor-spectrum', {
        display: "<div spectrum-colorpicker ng-model='color' on-change='!!color && action(color)' format='\"hex\"' options='options'></div>",
        action: function (color) {
            var me = this;
            if (!this.$editor().wrapSelection) {
                setTimeout(function () {
                    me.action(color);
                }, 100)
            } else {
                return this.$editor().wrapSelection('foreColor', color);
            }
        },
        options: {
            replacerClassName: 'fa fa-font', showButtons: false, showPaletteOnly: true
        },
        color: "#000"
    });

works fine ? i tried and don't work :/ it reset input`

**Comment by Xiphe** [comment]
the code above seems not to be related to this library

**Comment by GuillaumeGrandjean** [comment]
Yes ok.. But i used your library with bower install angular-spectrum-colorpicker and I have the same problem... But well, maybe i did an error

**Comment by Xiphe** [comment]
i'd be happy to help there but the plunker you provided seems to not demonstrate the problem you're talking about. Which leaves me in a "can not reproduce" state

**Comment by GuillaumeGrandjean** [comment]
I can reproduce a plunker, but the one I gave you, in the first colorpicker which using directiver spectrum color-picker, when choosing a color for the 2nd time, the cursor returns to the beginning of the line, right? (or maybe it depends of navigator ..)

**Comment by Xiphe** [comment]
Ah i get it!

**Comment by GuillaumeGrandjean** [comment]
Nice, and you think you can figure out where the problem?

**Comment by Xiphe** [comment]
Not sure if I find time in the near future. I'd be happy to get some help here
